examples/dynamo/src/common/protect.ts

```diff
@@ -0,0 +1,121 @@
+import { protect, csColumn, csTable } from '@cipherstash/protect'
+import type { EncryptedPayload, ProtectClient } from '@cipherstash/protect'
+import type {
+  ProtectColumn,
+  ProtectTable,
+  ProtectTableColumn,
+} from '@cipherstash/protect'
+
+export const users = csTable('users', {
+  email: csColumn('email').equality(),
+})
+
+export const protectClient = await protect(users)
+
+export async function encryptModel(
+  item: Record<string, unknown>,
+  protectTable: ProtectTable<ProtectTableColumn>,
+): Promise<Record<string, unknown>> {
+  const encryptResult = await protectClient.encryptModel(item, protectTable)
+
+  if (encryptResult.failure) {
+    throw new Error(`encryption error: ${encryptResult.failure.message}`)
+  }
+
+  const data = encryptResult.data
+
+  const encryptedAttrs = Object.keys(protectTable.build().columns)
+
+  const encryptedItem = Object.entries(data).reduce(
+    (putItem, [attrName, attrValue]) => {
+      if (encryptedAttrs.includes(attrName)) {
+        const encryptPayload = attrValue as EncryptedPayload
+
+        putItem[`${attrName}__hmac`] = encryptPayload!.hm!
+        putItem[`${attrName}__source`] = encryptPayload!.c!
```

Comment:
The encryption/decryption funcs here currently expect the HMAC attr (for search) to be suffixed with `__hmac` and for the source ciphertext (for decryption) to be suffixed with `__source`.

We might want to allow users to provide names for these. For example, someone might want to name the attr `sk` if it's used in a search key.

Reply:
Similarly, we might want to add more flexibility around the key structure. Right now the value of the HMAC field is just the HMAC, but we might want to allow users to prefix their keys (e.g. add `EMAIL#` in front of the HMAC).

Reply:
Tracing issue here https://github.com/cipherstash/protectjs/issues/148